examples/scan.js

```diff
@@ -0,0 +1,34 @@
+var redis  = require("redis"),
+    client = redis.createClient();
+
+var cursor = 0;
+
+function scan() {
+    client.scan(
+        cursor,
+        "MATCH", "q:job:*",
+        "COUNT", "10",
+        function(err, res) {
+            if (err) throw err;
+
+            // Update the cursor position for the next scan
+            cursor = res[0];
+
+            // From <http://redis.io/commands/scan>:
+            // An iteration starts when the cursor is set to 0,
+            // and terminates when the cursor returned by the server is 0.
+            if (cursor === 0) {
+                return console.log('Iteration complete');
+            } else {
+                // Remember, more keys than COUNT or no keys may be returned
+                // See http://redis.io/commands/scan#the-count-option
+                if (res[1].length > 0) {
+                    return console.log('Array of matching keys', res[1]);
```

Comment:
it seems that in this case, scan() wouldn't be called again, so we'd only get the first batch of items out of Redis, and we wouldn't get the full set of keys that match "q:job:*".

I feel it'd be better to do something like:

```javascript
if (res[1].length > 0) {
  console.log('Array of matching keys', res[1]);
}
return scan();
```